core/src/main/scala/org/http4s/blaze/channel/nio1/NIO1SocketServerGroup.scala

```diff
@@ -0,0 +1,238 @@
+
+        // Add any new connections
+        val q = queue.getAndSet(Nil)
+
+        q.foreach { ch =>
+          try {
+            ch.channel.configureBlocking(false)
+            ch.channel.register(s, SelectionKey.OP_ACCEPT, ch)
+
+          } catch {
+            case NonFatal(t) =>
+              logger.error(t)("Error during channel registration: " + ch.channel.getLocalAddress())
+              try ch.close()
+              catch { case NonFatal(_) => /* NOOP */ }
+          }
+        }
+
+        val it = s.selectedKeys().iterator()
+
+        while (it.hasNext()) {
+          val key = it.next()
+          it.remove()
+
+          val channel = key.attachment().asInstanceOf[NIO1ServerChannel]
+          val serverChannel = channel.channel
+          val service = channel.service
+          val loop = pool.nextLoop()
+
+          try {
+            val clientChannel = serverChannel.accept()
+
+            if (clientChannel != null) {                       // This should never be `null`
+            val address = clientChannel.getRemoteAddress()
+
+              // check to see if we want to keep this connection
+              if (acceptConnection(address)) {
+                clientChannel.setOption(java.net.StandardSocketOptions.TCP_NODELAY, java.lang.Boolean.FALSE)
+                loop.initChannel(service, clientChannel, key => new SocketChannelHead(clientChannel, loop, key))
+              }
+              else clientChannel.close()
+            }
+          }
+          catch {
+            case s: SecurityException => logger.info(s)("Connection rejected by SecurityManager.")
+
+            case e: ClosedChannelException =>
+              logger.warn(s"Channel ${serverChannel.getLocalAddress()} is closed.")
+              key.cancel()
+              channel.close()
+
+            case e: IOException =>
+              val localAddress = serverChannel.getLocalAddress()
+              logger.warn(e)(s"Error accepting connection on address $localAddress")
+
+              // If the server channel cannot go on, disconnect it.
+              if (!serverChannel.isOpen()) {
+                logger.warn(s"Channel for address $localAddress")
+                key.cancel()
+                channel.close()
+              }
+
+            case t: Throwable =>
+              logger.error(t)("Fatal error in selector. Closing Group")
+              closeGroup() // will cause the closing of the attached channels
+          }
+        }
+      }
+
+      // We have been closed. Close all the attached channels as well
+      val it = s.keys().iterator()
+      while (it.hasNext()) {
+        val key = it.next()
+        val ch = key.attachment().asInstanceOf[NIO1ServerChannel]
+        key.cancel()
+        ch.close()
+
+      }
+
+      // Close down the selector loops
+      pool.shutdown()
+
+      // clear out the queue
+      queue.getAndSet(null).foreach { ch =>
+        try ch.close()
+        catch { case NonFatal(_) => /* NOOP */ }
```

Comment:
I might log.debug these as a compromise between hiding harmless exceptions and being wrong about what's harmless.